### PR TITLE
👽️ 🌐 513 - step 2 UI hookup for GET

### PR DESCRIPTION
**Pull request by anncatton**
## Description of Changes

Adds a `GET` request to connect Wizard Step 2 to the Consent API. `POST` request will be handled in a separate PR.

### Consent UI
- adds API request and handler function for retrieving ConsentReleaseData from the `/consent-release-data` Consent API endpoint
- updates form `name` accessors to match schema changes
- adds dictionaries for `PrimaryCancerDiagnosis` to support the `selfReportedPrimaryCancerDiagnosis` field. The dictionary values will use the full string text from Data Model definition set.
- adds `selfReportedPrimaryCancerDiagnosis` options to the form dropdown

### Types
- adds a `nullable` -> `undefined` transform for `participantPreferredName` in `PIUser`, looks like I missed this in the `/user` endpoint implementation

### Special Instructions
- From `types` package, run `pnpm build` to pick up the new type defs. 
- To test, must be logged in with a registered Participant (so your session contains a `participantId`

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [ ] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/consent-ui/src/components/views/ConsentWizard/ConsentReleaseData/ConsentReleaseDataForm/index.tsx [hunk 1]
```diff
@@ -215,10 +221,10 @@ const ConsentReleaseDataForm = ({
 					<SelectFieldSet
 						label={labelsDict.primaryCancerDiagnosis}
 						description={textDict.primaryCancerDiagnosisTooltip}
-						name="primaryCancerDiagnosis"
-						error={undefined} // TODO: Add primaryCancerDiagnosis to zod schema
```
Comment (mistryrn) [inline apps/consent-ui/src/components/views/ConsentWizard/ConsentReleaseData/ConsentReleaseDataForm/index.tsx:219]: 🎉

File: apps/consent-ui/src/components/views/ConsentWizard/ConsentReleaseData/ConsentReleaseDataForm/index.tsx [hunk 2]
```diff
@@ -215,10 +221,10 @@ const ConsentReleaseDataForm = ({
...
+						name="selfReportedPrimaryCancerDiagnosis"
+						error={errors.selfReportedPrimaryCancerDiagnosis?.type && errorsDict.required}
 						placeholder={textDict.selectPlaceholder}
-						options={[]} // TODO: Add cancer options
```
Comment (mistryrn) [inline apps/consent-ui/src/components/views/ConsentWizard/ConsentReleaseData/ConsentReleaseDataForm/index.tsx:221]: 🎉

File: apps/consent-ui/src/i18n/locales/en/primaryCancerDiagnosis.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	BREAST: 'Breast',
+	BRONCHUS_AND_LUNG: 'Bronchus and lung',
+	CERVIX_UTERI: 'Cervix uteri',
+	COLON: 'Colon',
+	CONNECTIVE_SUBCUTANEOUS_AND_OTHER_SOFT_TISSUES: 'Connective, subcutaneous and other soft tissues',
+	CORPUS_UTERI: 'Corpus uteri',
+	ESOPHAGUS: 'Esophagus',
+	EYE_AND_ADNEXA: 'Eye and adnexa',
+	FLOOR_OF_MOUTH: 'Floor of mouth',
+
```
Comment (mistryrn) [inline apps/consent-ui/src/i18n/locales/en/primaryCancerDiagnosis.ts:41]: Not a blocker for approval/merging, but maybe in the next PR we can remove this line of whitespace?

File: apps/consent-ui/src/i18n/locales/fr/primaryCancerDiagnosis.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+	BREAST: 'Breast',
+	BRONCHUS_AND_LUNG: 'Bronchus and lung',
+	CERVIX_UTERI: 'Cervix uteri',
+	COLON: 'Colon',
+	CONNECTIVE_SUBCUTANEOUS_AND_OTHER_SOFT_TISSUES: 'Connective, subcutaneous and other soft tissues',
+	CORPUS_UTERI: 'Corpus uteri',
+	ESOPHAGUS: 'Esophagus',
+	EYE_AND_ADNEXA: 'Eye and adnexa',
+	FLOOR_OF_MOUTH: 'Floor of mouth',
+
```
Comment (mistryrn) [inline apps/consent-ui/src/i18n/locales/fr/primaryCancerDiagnosis.ts:41]: Similar whitespace comment as above

File: apps/consent-ui/src/services/api/requests/getConsentReleaseData.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+export const getConsentReleaseData = async (
+	session: Session | null,
+): HandledResponse<ConsentReleaseDataResponse> => {
+	const result = await consentApiFetch({
+		method: 'GET',
+		session,
+		url: API.CONSENT_RELEASE_DATA,
+	}).catch(catchAxiosError);
+
+	return result;
```
Comment (mistryrn) [inline apps/consent-ui/src/services/api/requests/getConsentReleaseData.ts:42]: So clean!

**Review by mistryrn — APPROVED** [review]
Looks good! A couple extra lines of whitespace in some of the enums, but we can clean that up later, no need to block the merging of this 🙂

File: apps/consent-ui/src/i18n/locales/en/primaryCancerDiagnosis.ts (hunk 3 above)
Comment (anncatton) [reply apps/consent-ui/src/i18n/locales/en/primaryCancerDiagnosis.ts:41]: whoops yes i'll fix this in the POST hookup pr

**Pull request merged**